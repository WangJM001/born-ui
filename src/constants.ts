export const CLASS_NAME_PREFIX = 'born-ui';
