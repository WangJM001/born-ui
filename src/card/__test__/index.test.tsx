import React from 'react';
import { mount } from 'enzyme';
import Card from '..';
import mountTest from '../../../tests/shared/mountTest';
import { CLASS_NAME_PREFIX } from '../../constants';

describe('Card', () => {
  mountTest(Card);

  it('should show header bottom border correctly', () => {
    const wrapper = mount(<Card title="card-title">Content</Card>);
    expect(
      wrapper.find(`Card.${CLASS_NAME_PREFIX}-card.${CLASS_NAME_PREFIX}-card-without-header-split`),
    ).toHaveLength(1);

    wrapper.setProps({ headerSplit: true });
    expect(
      wrapper.find(`Card.${CLASS_NAME_PREFIX}-card.${CLASS_NAME_PREFIX}-card-without-header-split`),
    ).toHaveLength(0);

    expect(wrapper.render()).toMatchSnapshot();
  });

  it('should show ghost mode correctly', () => {
    const wrapper = mount(<Card title="card-title">Content</Card>);
    expect(
      wrapper.find(`Card.${CLASS_NAME_PREFIX}-card.${CLASS_NAME_PREFIX}-card-ghost`),
    ).toHaveLength(0);
    wrapper.setProps({ ghost: true });
    expect(
      wrapper.find(`Card.${CLASS_NAME_PREFIX}-card.${CLASS_NAME_PREFIX}-card-ghost`),
    ).toHaveLength(1);

    expect(wrapper.render()).toMatchSnapshot();
  });

  it('should show border correctly', () => {
    const wrapper = mount(<Card title="card-title">Content</Card>);
    expect(wrapper.find(`.${CLASS_NAME_PREFIX}-card.ant-card-bordered`)).toHaveLength(0);

    wrapper.setProps({ bordered: true });
    expect(wrapper.find(`.${CLASS_NAME_PREFIX}-card.ant-card-bordered`)).toHaveLength(1);

    expect(wrapper.render()).toMatchSnapshot();
  });
});
