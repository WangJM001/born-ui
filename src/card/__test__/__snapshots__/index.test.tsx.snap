// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should show border correctly 1`] = `
<div
  class="ant-card ant-card-bordered born-ui-card born-ui-card-without-header-split"
>
  <div
    class="ant-card-head"
  >
    <div
      class="ant-card-head-wrapper"
    >
      <div
        class="ant-card-head-title"
      >
        card-title
      </div>
    </div>
  </div>
  <div
    class="ant-card-body"
  >
    Content
  </div>
</div>
`;

exports[`Card should show ghost mode correctly 1`] = `
<div
  class="ant-card born-ui-card born-ui-card-ghost born-ui-card-without-header-split"
>
  <div
    class="ant-card-head"
  >
    <div
      class="ant-card-head-wrapper"
    >
      <div
        class="ant-card-head-title"
      >
        card-title
      </div>
    </div>
  </div>
  <div
    class="ant-card-body"
  >
    Content
  </div>
</div>
`;

exports[`Card should show header bottom border correctly 1`] = `
<div
  class="ant-card born-ui-card"
>
  <div
    class="ant-card-head"
  >
    <div
      class="ant-card-head-wrapper"
    >
      <div
        class="ant-card-head-title"
      >
        card-title
      </div>
    </div>
  </div>
  <div
    class="ant-card-body"
  >
    Content
  </div>
</div>
`;
